### PR TITLE
Speed ups

### PR DESCRIPTION
**Pull request by jridgewell**
So we can discuss.

### CODE REVIEW

File: underscore.js [hunk 1]
```diff
@@ -96,21 +96,26 @@
 
   // Similar to ES6's rest params (http://ariya.ofilabs.com/2013/03/es6-and-rest-parameter.html)
   // This accumulates the arguments passed into an array, after a given index.
-  _.restParams = function rest(func, startIndex) {
-    startIndex = startIndex != null ? +startIndex : 1;
+  _.restParams = function(func) {
+    var startIndex = func.length - 1;
```
Comment (jridgewell) [inline underscore.js]: What sense would it be to have it be anything but the last arg? The whole method would have to be rewritten to support it any place else.

Comment (megawac) [inline underscore.js]: Consider something like

``` js
var restArgs = _.restParams(_.rest, 0);
restArgs(1,2,3,4,5,6,7);
```

Comment (jridgewell) [inline underscore.js]: ~~Won't that lose `guard`?~~ I see what you're saying now.

We could default...

``` javascript
_.restParams = function(func, startIndex) {
  startIndex = startIndex == null ? func.length - 1 : startIndex
};
```

File: underscore.js [hunk 2]
```diff
@@ -720,15 +725,14 @@
   // Create a function bound to a given object (assigning `this`, and arguments,
   // optionally). Delegates to **ECMAScript 5**'s native `Function.bind` if
   // available.
-  _.bind = function(func, context) {
+  _.bind = _.restParams(function(func, context, args) {
```
Comment (megawac) [inline underscore.js]: This will break the nativeBind case :heavy_multiplication_x:

File: underscore.js (hunk 1 above)
Comment (megawac) [inline underscore.js]: Yeah that's reasonable.

File: underscore.js (hunk 2 above)
Comment (jridgewell) [inline underscore.js]: Fixed.

File: underscore.js (hunk 1 above)
Comment (jridgewell) [inline underscore.js]: Fixed.

File: underscore.js (hunk 2 above)
Comment (megawac) [inline underscore.js]: That `restParams` is still going to break things

Comment (jridgewell) [inline underscore.js]: Really, really fixed it this time.

**Pull request closed without merging**